### PR TITLE
Fix wrong user/password state misrepresented as expired session

### PR DESCRIPTION
**Pull request by nirvn**
@mbernasocchi , @suricactus , as discussed.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for af7cc7b4ddc98c061b8bd959f03f7f858bf7d71b: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-arm-neon-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-arm64-android.apk)
- [all access arm-neon-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-arm-neon-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4660-2fcdbfea9f3e09a9a61bd56c4cbe741ea45b06de-x86-android.apk)
</details>

File: src/core/qfieldcloudconnection.cpp [hunk 1]
```diff
@@ -181,14 +182,16 @@ void QFieldCloudConnection::login()
...
       }
-      else if ( httpCode == 400 )
+      else if ( httpCode == 400 || httpCode == 401 )
       {
-        emit loginFailed( tr( "Wrong username or password" ) );
-      }
-      else if ( httpCode == 401 )
-      {
-        emit loginFailed( tr( "Session expired" ) );
-        invalidateToken();
```
Comment (suricactus) [inline src/core/qfieldcloudconnection.cpp:191]: Is it ok to skip the `invalidateToken`?

**Review by suricactus — COMMENTED** [review]
I like the approach. Thanks!

**Review by suricactus — APPROVED** [review]
I like the approach. Thanks!

File: src/core/qfieldcloudconnection.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/qfieldcloudconnection.cpp:191]: @suricactus , yes, it's invalidated by the get() function (https://github.com/opengisch/QField/blob/master/src/core/qfieldcloudconnection.cpp#L380-L387)

Comment (suricactus) [reply src/core/qfieldcloudconnection.cpp:191]: Ok, thanks!

**Review by suricactus — APPROVED** [review]

**Pull request merged**